### PR TITLE
dtool: init at 0.12.0

### PR DESCRIPTION
**Pull request by linuxissuper**
###### Description of changes

`dtool` is a command-line tool collection to assist development
from https://github.com/guoxbin/dtool

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 226460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dtool</li>
  </ul>
</details>

**Review by Janik-Haag — COMMENTED** [review]
`dtool usage` throws this error; `Segmentation fault (core dumped)` but `dtool ue` works, do you know why this is?

**Comment by linuxissuper** [comment]
hmm I don't understand   
when I build it in `pkgs/tools/misc/dtool`   
with `nix-build -K -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`  it works   
  
but when building it with `nix-build -A dtool` it does not 🤔

**Comment by linuxissuper** [comment]
`nix-build` itself never throws an error  
but with `nix-build -A dtool` in nixpkgs root, I get the same error you mentioned:  
`zsh: segmentation fault (core dumped)  ./result/bin/dtool usage`   
 
according to `coredumpctl` `prettytable-rs` seems to be the problem   
but I don't know why

**Comment by Janik-Haag** [comment]
Hm you could try packaging prettytable-rs and check if that works. I sadly don't know much about rust.

**Comment by Janik-Haag** [comment]
Maybe ask in the nixpkgs matrix channel for help since I have no idea why this happens.

**Comment by linuxissuper** [comment]
I converted the PR to draft  
Did not try to fix this since last time  
  
I'll try to find some time to fix this

**Comment by linuxissuper** [comment]
It works now (I actually still don't know what was the problem and why it is working now... 🙈)

File: pkgs/tools/misc/dtool/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description = "A command-line tool collection to assist development written in RUST";
+    homepage = "https://github.com/guoxbin/dtool";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ ];
```
Comment (Janik-Haag) [inline pkgs/tools/misc/dtool/default.nix:29]: you should add your self as a maintainer. If you don't already have a maintainer entry add one in https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix and make it it's on commit called `maintainers: add yourName`

**Review by Janik-Haag — COMMENTED** [review]
Darwin tests are currently failing you should probably disable the failing ones or mark the platform as broken. Other then that this looks good.

File: pkgs/tools/misc/dtool/default.nix (hunk 1 above)
Comment (linuxissuper) [reply pkgs/tools/misc/dtool/default.nix]: I've added myself

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**